backend/cfg/cfg_dominators.ml

```diff
@@ -413,12 +413,65 @@ let compute_dominator_forest : Cfg.t -> doms -> dominator_tree list =
   if debug then invariant_dominator_forest cfg doms res;
   res
 
+let compute_dominator_forest_opt : Cfg.t -> doms -> dominator_tree list =
+ fun cfg doms ->
+  let roots = ref [] in
+  let children = Label.Tbl.create (Label.Tbl.length cfg.blocks) in
+  let rec build_tree (label : Label.t) : dominator_tree =
+    let children_labels =
+      match Label.Tbl.find_opt children label with
+      | None -> []
+      | Some labels -> labels
+    in
+    { label; children = List.rev_map children_labels ~f:build_tree }
+  in
+  Cfg.iter_blocks cfg ~f:(fun label _block ->
+      match Label.Tbl.find_opt doms label with
+      | None -> assert false
+      | Some immediate_dominator ->
+        if Label.equal label immediate_dominator
+        then roots := label :: !roots
+        else
+          let current =
+            match Label.Tbl.find_opt children immediate_dominator with
+            | None -> []
+            | Some children -> children
+          in
+          Label.Tbl.replace children immediate_dominator (label :: current));
+  let res = List.rev_map !roots ~f:build_tree in
```

Comment:
Where do we rely on the order of roots and children in the tree? Why do the lists need to be reversed in L441  and L426?

Reply:
We don't, that is actually an unfortunate leftover,
due to a try to reproduce an unrelated bug. Thanks!